### PR TITLE
ensure all go tests build

### PR DESCRIPTION
**Pull request by dprotaso**

### CODE REVIEW

File: workflow-templates/knative-go-build.yaml [hunk 1]
```diff
@@ -52,3 +52,6 @@ jobs:
 
       - name: Build
         run: go build -v ./...
+
+      - name: Build Test
+        run: for d in $(go list -tags e2e -f '{{if .TestGoFiles}}{{.Dir}}{{end}}' ./... ); do (echo $d; cd $d && go test -tags e2e -c ); done
```
Comment (mattmoor) [inline workflow-templates/knative-go-build.yaml:57]: We have a bunch more tags (e.g. `performance`, `postupgrade`, `preupgrade`) I think there's a script in our wad of bash to extract and pass them all.

Comment (mattmoor) [reply workflow-templates/knative-go-build.yaml:57]: except the one we use for `hack/tools.go` IIRC

**Comment by dprotaso** [comment]
looks like `vet` catches issues in test files - so it'll be just `_test` files with build tags

**Comment by dprotaso** [comment]
/hold cancel

`vet` with tags reports build failures in `test` files as well

**Comment by dprotaso** [comment]
i forgot the third_party code

**Comment by mattmoor** [comment]
/close

Can you send this to `knative-sandbox`?  This is just a time-delayed remote of that

**Pull request closed without merging**